### PR TITLE
Expose patricia_search_all on SubnetTree

### PR DESCRIPTION
**Pull request by ikbenale**
Note: there are some unrelated changes on `SubnetTree_wrap.cc` that appear when I `make` the project. First I assumed it was because I was using a newer version of SWIG, but I downloaded and compiled the one used in the file comments, 3.0.12, and I still got those discrepancies. If it's more convenient, I can manually edit those changes to omit unwanted lines.

### CODE REVIEW

**Comment by bbannier** [comment]
> Note: there are some unrelated changes on SubnetTree_wrap.cc that appear when I make the project.

Yeah, I saw that to, there must have been something else about these files. The [commit making the last update](https://github.com/zeek/pysubnettree/commit/9af28ed6e67e6a76af31302c0acfed4c26480c03) talks about keeping changes minimal, maybe there was some manual editing involved then.

For fixing a new runtime failure with e.g., python-3.11.9 we have merged a fix https://github.com/zeek/pysubnettree/pull/38 which updates the generated files so you should see a smaller diff now. That PR also brings CI up to date (e.g., python-3.7 is not available anymore in macos CI). If you rebase your PR on the lasted `HEAD` you'll get these changes.

**Comment by ikbenale** [comment]
@bbannier rebased and now my diff looks clean 👍. I pushed the new version.

**Comment by ngbtri** [comment]
Am I right to say `search-all` will return all CIDRs which an IP belongs to, not just the last leaf?
If yes, this is exciting!

File: SubnetTree.cc [hunk 1]
```diff
@@ -300,6 +300,51 @@ PyObject* SubnetTree::lookup(int family, inx_addr addr) const
     return data;
 }
 
+PyObject* SubnetTree::search_all(const char *cidr) const
```
Comment (awelzel) [inline SubnetTree.cc:303]: For completeness, should this support binary mode like `lookup()` does?

File: testing/pysubnettree/search-all.test [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+def ipv6_func(ipstr):
+    return ipstr
+
+def do_tests(binarymode):
+    if binarymode:
+        v4 = ipv4binary
+        v6 = ipv6binary
+    else:
+        v4 = ipv4_func
+        v6 = ipv6_func
```
Comment (awelzel) [inline testing/pysubnettree/search-all.test:20]: This test doesn't use `v4` or `v6`, so this is causing confusion, but see comment about implementing binarymode in which case you might want to use them.

**Review by awelzel — CHANGES_REQUESTED** [review]

File: SubnetTree.cc (hunk 1 above)
Comment (ikbenale) [reply SubnetTree.cc:303]: @awelzel would we need a `size` param in that case, like with `lookup`, to determine the family?

Comment (awelzel) [reply SubnetTree.cc:303]: I believe so. Basically, when in binarymode=True mode, `search_all()` with a bytestring from `inet_pton()` should work, too:

```
In [2]: socket.inet_pton(socket.AF_INET, "192.168.0.1")
Out[2]: b'\xc0\xa8\x00\x01'
```

Comment (ikbenale) [reply SubnetTree.cc]: @awelzel pushed a new commit to support binary mode. Let me know how that looks to you.

**Comment by awelzel** [comment]
@ikbenale - I've merged after adding a `free()` for the returned outlist to pluck a memory leak.

Thanks for your contribution!

**Comment by ikbenale** [comment]
@awelzel awesome, thanks!

>  I've merged after adding a free()

Out of curiosity, where did you add this? I don't see that change on `master` 🤔

**Comment by awelzel** [comment]
The change is in the merge commit. This part look slightly different now:

https://github.com/zeek/pysubnettree/blob/ebca8eddce0b654ecd3f5e354038a28f9a163a0f/SubnetTree.cc#L353-L360

**Pull request merged**